share/smack/prelude.py

```diff
@@ -0,0 +1,290 @@
+    bpl += map(lambda size: inline_binary_pred(bit_width = size, base_name = '$' + op).toStr(), sizes)
+
+  for i, op in enumerate(eq_ops):
+    bpl += map(lambda size: inline_binary_bv_comp(bit_width = size, base_name = '$' + op, body = eq_ops_body[i]).toStr(), sizes)
+    bpl += map(lambda size: inline_binary_bv_pred(bit_width = size, base_name = '$' + op).toStr(), sizes)
+    bpl += map(lambda size: inline_binary_comp(bit_width = size, base_name = '$' + op, body = eq_ops_body[i]).toStr(), sizes)
+    bpl += map(lambda size: inline_binary_pred(bit_width = size, base_name = '$' + op).toStr(), sizes)
+
+  bpl += map(lambda size: inline_bvbuiltin_binary_select(bit_width = size, base_name = '$min', base_name_2 = '$slt').toStr(), sizes)
+  bpl += map(lambda size: inline_bvbuiltin_binary_select(bit_width = size, base_name = '$max', base_name_2 = '$sgt').toStr(), sizes)
+  bpl += map(lambda size: inline_bvbuiltin_binary_select(bit_width = size, base_name = '$umin', base_name_2 = '$ult').toStr(), sizes)
+  bpl += map(lambda size: inline_bvbuiltin_binary_select(bit_width = size, base_name = '$umax', base_name_2 = '$ugt').toStr(), sizes)
+
+  bpl += map(lambda size: inline_binary_op(bit_width = size, base_name = '$smin', body = '$min(i1,i2)').toStr(), sizes)
+  bpl += map(lambda size: inline_binary_op(bit_width = size, base_name = '$smax', body = '$max(i1,i2)').toStr(), sizes)
+  bpl += map(lambda size: inline_binary_op(bit_width = size, base_name = '$umin', body = '$min(i1,i2)').toStr(), sizes)
+  bpl += map(lambda size: inline_binary_op(bit_width = size, base_name = '$umax', body = '$max(i1,i2)').toStr(), sizes)
+
+  bpl += map(lambda size: bvbuiltin_unary_op(bit_width = size, builtin_name = 'bvnot', base_name = '$not').toStr(), sizes)
+  bpl += map(lambda size: uninterpreted_unary_op(bit_width = size, base_name = '$not').toStr(), sizes)
+
+  bpl += map(lambda size_pair: trunc_bv_op(bit_width = size_pair[1], bit_width_2 = size_pair[0]).toStr(), itertools.combinations(sizes, 2))
+  bpl += map(lambda size_pair: trunc_op(bit_width = size_pair[1], bit_width_2 = size_pair[0]).toStr(), itertools.combinations(sizes, 2))
+
+  bpl += map(lambda size_pair: inline_conversion(bit_width = size_pair[0], bit_width_2 = size_pair[1], base_name = '$zext', body = 'i').toStr(), itertools.combinations(sizes, 2))
+  bpl += map(lambda size_pair: inline_conversion(bit_width = size_pair[0], bit_width_2 = size_pair[1], base_name = '$sext', body = 'i').toStr(), itertools.combinations(sizes, 2))
+
+  bpl += map(lambda size: safe_load_op(bit_width = size, body = 'M[p]').toStr(), sizes)
+  bpl += map(lambda size: safe_load_bv_op(bit_width = size, body = 'M[p]').toStr(), sizes)
+  bpl += map(lambda size: safe_store_op(bit_width = size, body = 'M[p := v]').toStr(), sizes)
+  bpl += map(lambda size: safe_store_bv_op(bit_width = size, body = 'M[p := v]').toStr(), sizes)
+
+  return ''.join(bpl)
```

Comment:
You could get rid of all of the previous calls to `toStr` by replacing this line with:
```Python
return ' '.join([str(x) for x in bpl])
```